frontend/packages/dev-console/src/components/monitoring/metrics/MetricsChart.tsx

```diff
@@ -0,0 +1,56 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { queryBrowserDeleteAllQueries } from '@console/internal/actions/ui';
+import { getActiveNamespace } from '@console/internal/reducers/ui';
+import { QueryBrowser, QueryObj } from '@console/internal/components/monitoring/query-browser';
+import { QueryTable } from '@console/internal/components/monitoring/metrics';
+import EmptyStateQuery from './EmptyStateQuery';
+
+interface StateProps {
+  queries: QueryObj[];
+  namespace: string;
+}
+type MetricsChartProps = {
+  deleteAll: () => never;
+  queries: QueryObj[];
+  namespace: string;
+};
+const MetricsChart: React.FC<MetricsChartProps> = ({ deleteAll, queries, namespace }) => {
+  React.useEffect(() => deleteAll, [deleteAll]);
+  /* eslint-disable react-hooks/exhaustive-deps */
+  // Use React.useMemo() to prevent these two arrays being recreated on every render, which would trigger unnecessary
+  // re-renders of QueryBrowser, which can be quite slow
+  const queriesMemoKey = JSON.stringify(_.map(queries, 'query'));
+  const queryStrings = React.useMemo(() => _.map(queries, 'query'), [queriesMemoKey]);
+  const disabledSeriesMemoKey = JSON.stringify(
+    _.reject(_.map(queries, 'disabledSeries'), _.isEmpty),
+  );
+  const disabledSeries = React.useMemo(() => _.map(queries, 'disabledSeries'), [
+    disabledSeriesMemoKey,
+  ]);
```

Comment:
If we are using this react.memo approach  to improve the performance and `eslint-disable` comment to avoid get lint errors, then its a good practice to  enable it back after the usage.  
```suggestion
  ]);
   /* eslint-enable react-hooks/exhaustive-deps */
```

Reply:
Only use eslint-disable-next-line